survival/client/survival_analysis.go

```diff
@@ -0,0 +1,262 @@
+	if err != nil {
+		return
+	}
+
+	return
+
+}
+
+// EncryptedResults holds a TimePoint and the corresponding encrypted events
+type EncryptedResults struct {
+	TimePoint int
+	Events    Events
+}
+
+//Events holds the number of events at a given timepoint in base64 strings
+type Events struct {
+	EventsOfInterest string
+	CensoringEvents  string
+}
+
+//Decrypt deciphers a value that is expected to be encrypted under user public key
+func (clientSurvivalAnalysis *SurvivalAnalysis) Decrypt(value string) (int64, error) {
+	return unlynx.Decrypt(value, clientSurvivalAnalysis.userPrivateKey)
+}
+
+type NodeResult struct {
+	Body      *survival_analysis.SurvivalAnalysisOKBody
+	NodeIndex int
+}
+
+//Execute is the main function that sends the request and waits
+func (clientSurvivalAnalysis *SurvivalAnalysis) Execute() (results map[string][]*EncryptedResults, err error) {
+	totalTimer := time.Now()
+	defer func(since time.Time) {
+		err = clientSurvivalAnalysis.addTimer("time for the total execution ", since)
+	}(totalTimer)
+	errChan := make(chan error)
+	resultChan := make(chan NodeResult)
+
+	for idx := range clientSurvivalAnalysis.httpMedCoClients {
+
+		go func(idx int) {
+			res, Error := clientSurvivalAnalysis.submitToNode(idx)
+			if Error != nil {
+				logrus.Errorf("Survival analysis exection error : %s", Error)
+				errChan <- Error
+			} else {
+
+				resultChan <- NodeResult{Body: res, NodeIndex: idx}
+			}
+		}(idx)
+	}
+	//TODO magic number
+	timeout := time.After(time.Duration(3000) * time.Second)
+
+	results = make(map[string][]*EncryptedResults)
+nodeLoop:
+	for idx := range clientSurvivalAnalysis.httpMedCoClients {
+		select {
+		case nodeLoopErr := <-errChan:
+			if err != nil {
+				err = fmt.Errorf("Node %d threw %s : %s", idx, nodeLoopErr, err)
+			} else {
+				err = fmt.Errorf("Node %d threw %s", idx, nodeLoopErr)
+			}
+		case nodeLoopRes := <-resultChan:
+			for label, value := range nodeLoopRes.Body.Timers {
+
+				timerErr := clientSurvivalAnalysis.profiling.Write([]string{label, strconv.FormatFloat(value, 'f', -1, 64), strconv.Itoa(nodeLoopRes.NodeIndex)})
+				if timerErr != nil {
+					err = fmt.Errorf("Node %d threw %s : %s", idx, timerErr, err)
+					return
+				}
+
+			}
+			clientSurvivalAnalysis.profiling.Flush()
+
+			for _, groups := range nodeLoopRes.Body.Results {
+
+				var innerList []*EncryptedResults
+				for _, val := range groups.GroupResults {
+					innerList = append(innerList, &EncryptedResults{TimePoint: int(val.Timepoint),
+						Events: Events{EventsOfInterest: val.Events.Eventofinterest,
+							CensoringEvents: val.Events.Censoringevent,
+						}})
+				}
+
+				results[groups.GroupID] = append(results[groups.GroupID], innerList...)
+
+			}
+
+		case <-timeout:
+			err = fmt.Errorf(" Timeout : %s", err.Error())
+			break nodeLoop
+
+		}
+	}
+
+	return
+}
+
+func (clientSurvivalAnalysis *SurvivalAnalysis) addTimer(label string, since time.Time) (err error) {
```

Comment:
As said in previous comment, timers implementation should be put in common with the explore query timers.